gurubu-client/src/app/gurubu-planner/components/SelectTeamForm.tsx

```diff
@@ -39,6 +38,90 @@ export const SelectTeamForm = ({
   const pService = new PService(process.env.NEXT_PUBLIC_API_URL || "");
   const jiraService = new JiraService(process.env.NEXT_PUBLIC_API_URL || "");
 
+  useEffect(() => {
+    const teamFromUrl = searchParams.get('team');
+    const boardFromUrl = searchParams.get('board');
+
+    if (teamFromUrl) setSelectedTeam(teamFromUrl);
+    if (boardFromUrl) setSelectedBoard(boardFromUrl);
+  }, [searchParams]);
+
+  useEffect(() => {
+    const fetchDataFromUrl = async () => {
+      const teamFromUrl = searchParams.get('team');
+      const boardFromUrl = searchParams.get('board');
+      const boardSearchFromUrl = searchParams.get('boardSearch');
+
+      if (teamFromUrl) {
+        const response = await pService.getOrganizationDetails(teamFromUrl);
+        if (response.isSuccess && response.data) {
+          setAssignees(response.data);
+          localStorage.setItem('JIRA_DEFAULT_ASSIGNEES', JSON.stringify(response.data));
+        }
+      }
+
+      if (boardSearchFromUrl) {
+        await fetchBoards(boardSearchFromUrl);
+      }
+
+      if (boardFromUrl) {
+        try {
+          const response = await fetch(`${process.env.NEXT_PUBLIC_API_URL}/jira/${boardFromUrl}/future`);
+          const data: SprintResponse = await response.json();
+          setSprints(data.values);
+          localStorage.setItem('JIRA_BOARD', boardFromUrl);
+          handleRefresh();
+        } catch (error) {
+          console.error('Error fetching future sprints:', error);
+        }
+      }
+    };
+
+    fetchDataFromUrl();
+  }, [searchParams]);
```

Comment:
Is searchParams constantly updated, for example, I wonder if this place will be hit every time I press a character.